### DISCUSSION TITLE
Access the coordinates of an element and neighbor inside an interface kernel

### DISCUSSION THREAD

**Discussion by ambehnam** [idaholab/moose#21510] [Q&A General] [answered]
Hi there. I need to access the coordinates of the element and neighbor inside an interface kernel to compute a penalty term. 

This is my code:
```
  const libMesh::Node* const* curr_node = _var.currentElem()->get_nodes();
  const libMesh::Node* const* neigh_node = _var.neighbor()->get_nodes();
  Real dv = curr_node[0]->_coords[0];
```
and this is the error I get:
```
error: ‘double libMesh::TypeVector<double>::_coords [3]’ is protected within this context
   61 |   Real dv = curr_node[0]->_coords[0];
```
Can anyone help me fix this issue please?

Thanks

**Comment by GiudGiud** [comment, marked as answer]
hello

You may not access _coords directly from a Node.
As you may see on the doxygen
https://libmesh.github.io/doxygen/classlibMesh_1_1Node.html
and as the compiler tells you, it's protected.

You need to access using the () operator
https://libmesh.github.io/doxygen/classlibMesh_1_1Node.html#a3568df25f9c0db5decbcaa9d42420439

Guillaume

**Reply by ambehnam** [reply]
Thanks a lot!

Amir

**Reply by jessecarterMOOSE** [reply]
@GiudGiud Wanted to follow up on this. How do you access the () operator on a node to get coordinates? `_current_node()` does not work. Will _q_point work if it's a nodal kernel?

**Reply by GiudGiud** [reply]
_current_node(0) should work? to get you the X coordinate

If q_point works for a nodal kernel it s out of trying to make every kernel look the same. But current_node is better

**Reply by jessecarterMOOSE** [reply]
Not working in my AuxKernel:
```
error: expression cannot be used as a function
  114 |   Real x = _current_node(0);
      |            ~~~~~~~~~~~~~^~~
```

**Reply by GiudGiud** [reply]
(*_current_node)(0) 

current node is a reference to a pointer, smh
```
  /// current node being processed
  const Node * const & _current_node;
```

**Reply by jessecarterMOOSE** [reply]
ah, there we go. So I need to dereference to use operator `(*_current_node)(0)` but other methods I can use directly `_current_node->id()`

**Reply by jessecarterMOOSE** [reply]
@GiudGiud what if _current_node isn't available? For instance I'm looping over nodes using _mesh.getLocalNodeRange() in my AuxKernel's initial setup.

**Reply by GiudGiud** [reply]
_current_node should be available in all auxkernels

See functionAux for an example of using it

**Reply by GiudGiud** [reply]
Or do you mean you nested a loop on all local nodes within an auxkernel ?

Ok that s your situation. I misread your initial post


In that case you can use the node from the loop? If you only have an id for the node in the loop, you can retrieve the node from the mesh with nodePtr (or similar from a meshBase)
